domain/src/test/java/org/oppia/android/domain/oppialogger/analytics/CpuPerformanceSnapshotterTest.kt

```diff
@@ -130,6 +136,7 @@ class CpuPerformanceSnapshotterTest {
     fakePerformanceMetricAssessor.setRelativeCpuUsage(TEST_CPU_USAGE_ONE)
     cpuPerformanceSnapshotter.updateAppIconification(APP_IN_FOREGROUND)
     testCoroutineDispatchers.runCurrent()
+    testCoroutineDispatchers.advanceTimeBy(foregroundCpuLoggingTimePeriodMillis)
```

Comment:
runCurrent shouldn't be needed on the line above with this line present. Ditto elsewhere in the test suite.

Reply:
It's actually needed in this case -- since the work is done on multiple threads, it's important to bring the current one (`SwitchIconification`) to an idle state and then call the `advanceTimeBy` to process the next commandMessage (`ScheduleTakeSnapshot`) which also happens to have a delay that we have to get to the end of. FWIW, I also tried removing `runCurrent()` and the tests started to fail.

Reply:
Ah interesting. This might be indicating a design flaw with advanceTimeBy as advanceUntilidle doesn't have this limitation. Good to note.